docs: add istio version compat table (#3268)
Kong/docs.konghq.com: app/kubernetes-ingress-controller/2.0.x/guides/getting-started-istio.md

combine the features of Kong for ingress traffic with Istio's mesh
functionality.

> A [version compatibility reference][compat] is available for this guide which
> shows the versions of {{site.kic_product_name}} and Istio which are tested for
> compatibility.

Reviewer: ```suggestion
See the [version compatibility reference][compat] for the
tested compatible versions of {{site.kic_product_name}} and Istio.
```
